packages/babel-traverse/src/path/modification.js

```diff
@@ -126,7 +126,10 @@ export function insertAfter(nodes) {
     if (Array.isArray(this.container)) {
       return this._containerInsertAfter(nodes);
     } else if (this.isStatementOrBlock()) {
-      if (this.node) nodes.unshift(this.node);
+      // Unshift current node if it's not an empty expression
+      if (this.node && (!this.isExpressionStatement() || this.node.expression != null)) {
```

Comment:
I thought so too, but couldn't come up with a test for that case. Basically we need something that results in:

```js
{
  code 
  code 
  ;  <--  empty statement here
}
```
Without a test/bug that exposes the above behavior, there's no guarantee such situation is possible. Currently, the behavior is caused by that line to `this.container['expression'] = null`, which is only invoked before `insertAfter`. I haven't seen anything line that before `insertBefore`. That said, I could definitely make that change to `insertBefore` for the extra caution. Let me know.